docs/architecture/grpc-for-wcf-developers/apm.md

```diff
@@ -1,11 +1,221 @@
 ---
 title: Application Performance Management - gRPC for WCF Developers
-description: TO BE WRITTEN
+description: Logging, metrics and tracing for ASP.NET Core gRPC applications
 author: markrendle
 ms.date: 09/02/2019
 ---
 
 # Application Performance Management
 
+In modern production environments like Kubernetes, it is very important to be able to monitor applications to ensure they are running optimally. Concerns like logging and metrics have never been more important. ASP.NET Core, including gRPC, has first-class support for producing and managing log messages and metrics data, as well as *tracing* data. This section will explore these areas in more details.
+
+## Logging vs Metrics
+
+Before looking at the implementation details, it is necessary to understand the difference between logging and metrics.
+
+Logging is concerned with text messages that record detailed information about things that have happened in the system. Log messages may include exception data like stack traces, or structured data that provides context about the message. Logging output is commonly written to a searchable text store.
+
+Metrics refers to numeric data that is designed to be aggregated and presented using charts and graphs in a dashboard that provides a view of the overall health and performance of an application. Metrics data can also be used to trigger automated alerts when a threshold is exceeded. Examples of metrics data include the time taken to process requests, the number of requests per second being handled by an instance of a service, or the number of failed requests on an instance.
+
+## Logging in ASP.NET Core gRPC
+
+ASP.NET Core provides support for logging out-of-the-box, in the form of the [Microsoft.Extensions.Logging package](https://www.nuget.org/packages/Microsoft.Extensions.Logging). This is included with the Web SDK so there is no need to install it manually. The gRPC framework writes its internal diagnostic logging messages to ASP.NET Core logging so they can be processed/stored along with the application's own messages.
```

Comment:
> This is included with the Web SDK so there is no need to install it manually

I really don't like the phrasing here. grpc-dotnet, by virtue of using the ASP.NET Core App Model uses the same logging abstractions that ASP.NET Core does. You still will configure need to configure a production logging sink that use these abstractions.

Reply:
I think the things worth mentioning are
- grpc uses Microsoft.Extensions.Logging abstraction
- grpc writes framework logs to the configured logger
- You can bring in additional logging sinks
- How to use the log in user code (which you've covered well in the next section)